WordPress/gutenberg
# Replace ImageSizeControl component with DimensionsTool component

## What?

This PR replaces to-be-deprecated `__experimentalImageSizeControl` component with `DimensionsTool` component inside the media-text and latest-posts block. 

## Why?

Issue was introduced by ticket #52223 . `__experimentalImageSizeControl` component is marked to be deprecated in the some future, makes sense to get rid of it.

## How?

Basically I opened up image block and started copy-pasting code from there. 

Biggest curve ball was the requirement to use new components `<ToolsPanel>` and `<ToolsPanelItem>`

## Testing Instructions

1. Open a post or page
2. Insert a media-text block and latest-posts block
3. Observe the right side column for new settings and play around with them

### Testing Instructions for Keyboard


## Screenshots or screencast 

![Screenshot 2023-08-20 at 18 39 22](https://github.com/WordPress/gutenberg/assets/62872075/dd96e487-e3ff-4e62-9c86-a4c416012e10)
![Screenshot 2023-08-20 at 18 39 53](https://github.com/WordPress/gutenberg/assets/62872075/c020fea5-00f1-4d08-a3f9-fcf389f34e3c)
![Screenshot 2023-08-20 at 18 40 16](https://github.com/WordPress/gutenberg/assets/62872075/4835b9f8-1878-4e9d-a771-f186bfe93924)

First screenshot is taken while latest-post block is active. Second screenshot is taken while latest-posts block is active and after clicking the three dots next to "Featured image" label. Last one is same as second screenshot but taken while media-text is active and instead label being "Featured image" it now was "Setting".
